src/main/scala/io/qbeast/spark/delta/DeltaQbeastSnapshot.scala

```diff
@@ -44,6 +46,22 @@ case class DeltaQbeastSnapshot(protected override val snapshot: Snapshot)
 
   private val metadataMap: Map[String, String] = snapshot.metadata.configuration
 
+  /**
+   * The current table properties of the snapshot.
+   *
+   * We filter out the revision, leaving only Revision ID.
+   *
+   * @return
+   */
+  override def loadProperties: mutable.Map[String, String] =
+    snapshot.getProperties.filter(prop => !prop._1.startsWith(MetadataConfig.revision))
```

Comment:
You can use `filterKeys` instead.

Reply:
mmm.. but that method is returning just the keys...

Reply:
done!